### DISCUSSION TITLE
3.18.10 release

### DISCUSSION THREAD

**Discussion by lfcnassif** [sepinf-inc/IPED#861] [General]
I'm preparing a 3.18.10 bug fix release. Current included changes are here: https://github.com/sepinf-inc/IPED/blob/3.18.x/ReleaseNotes.txt

@tc-wleite and @hauck-jvsh do you think any recent enhancements are safe enough to be included?

Maybe #829, #811, #626, #809?

**Comment by wladimirleite** [comment]
I think #811 and #809 are safe.
I wouldn't include #829. Although it is a trivial upgrade, it may have some unexpected behavior in other environments (SO / JRE).

**Comment by hauck-jvsh** [comment]
I think that #626 is also safe to be included.

**Comment by lfcnassif** [comment]
Just included #809, #811, #626. And also #710, #722 and #817.

@tc-wleite I was thinking about including #119 and #419, since they are very modular, with minor changes to Image/VideoThumbTask. What do you think?

**Reply by wladimirleite** [reply]
> @tc-wleite I was thinking about including #119 and #419, since they are very modular, with minor changes to Image/VideoThumbTask. What do you think?

Sounds great!

**Reply by lfcnassif** [reply]
I'm cherry picking. But just remembered we are still in Tika-1.20 in 3.18.x, what could bring OOM problems with PDFs. What do you think about configuring PDF thumb generation to use "external" process for now? Or leave it as "internal" as the feature will be flagged as [experimental] in configs?

**Reply by lfcnassif** [reply]
Or maybe we can upgrade just PDFBox, will try to do some regression tests...

**Reply by wladimirleite** [reply]
If upgrading PDFBox is not too risky/complex, it seems a good option, otherwise using "external" as default should be safer.

**Reply by lfcnassif** [reply]
I had already done this in the past. I'll use the same PDFBox version you tested on #119, after some regression tests for common strings searches.
